libs/core/table/directives/table-container.directive.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    ContentChild,
+    Directive,
+    ElementRef,
+    Input,
+    OnDestroy,
+    Renderer2,
+    booleanAttribute,
+    inject
+} from '@angular/core';
+import { ResizeObserverService } from '@fundamental-ngx/cdk/utils';
+import { FD_TOOLBAR } from '@fundamental-ngx/core/toolbar';
+import { Subscription } from 'rxjs';
+import { TableHeaderDirective } from './table-header.directive';
+
+@Directive({
+    selector: '[fdTableContainer], [fd-table-container]',
+    standalone: true
+})
+export class TableContainerDirective implements OnDestroy {
+    /** Whether the scroll is applied on the page and not on the table */
+    @Input({ transform: booleanAttribute })
+    outerScroll = false;
+
+    /** @hidden */
+    @ContentChild(FD_TOOLBAR, { read: ElementRef })
+    set toolbar(toolbarElRef: ElementRef) {
+        if (this.outerScroll && toolbarElRef) {
+            const toolbarEl = toolbarElRef.nativeElement;
+
+            if (this._resizeSubscription) {
+                this._resizeSubscription.unsubscribe();
+            }
+
+            this._resizeSubscription = this.resizeObserverService.observe(toolbarEl).subscribe(() => {
+                this._renderer.setStyle(toolbarEl, 'position', 'sticky');
```

Comment:
Why do we want to set all of these styles on a resize? A resize should be irrelevant to this in my opinion

Reply:
the position top of the Table header depends on the height of the Toolbar. In cozy mode the height of the toolbar is 2.75rem so the top of the table header should be 2.75rem. When we change the mode to compact, the height becomes 2rem so respectively the top of the Table header becomes 2rem. Initially I made a new subscription that was tracking the change of the toolbar height but saw that we already have one so I am using it to detect changes there and set the proper positions.